src/components/SnippetArea.jsx

```diff
@@ -131,54 +131,74 @@ class SnippetArea extends React.Component {
   startParserDaemon(parser) {
     setInterval(function() {
       const snippet = this.props.contents;
-      if (!snippet || snippet === this.state.prevSnippet) {
-        return;
-      }
-
-      // Generate an AST for the current state of the code snippet, if ready
-      const AST = parser(snippet);
-
-      // Get an array mapping token types to their occurence count
-      const tokenCount = [];
-      getTokenCount(AST, tokenCount);
-
-      // Generate array of strings containing pretty token name and its count
-      const filters = this.props.filters;
-      let newFilters = {};
-      Object.keys(tokenCount)
-        .filter(t => getPrettyTokenName(t) !== undefined)
-        .forEach(t => {
-          let selected = false;
-          if (filters[t]) {
-            selected = filters[t].selected;
-          }
-          newFilters[t] = {
-            prettyTokenName: getPrettyTokenName(t),
-            count: tokenCount[t],
-            selected,
-          }
-        });
-
-      // Highlight the code snippet and invoke prop callback
-      highlight(snippet, AST, this.codeMirror, newFilters);
-      this.setState({
-        prevSnippet: snippet
-      });
-      this.props.onParserRun(AST, newFilters);
+      if (snippet !== undefined &&
```

Comment:
This can be condensed down. Both `''` and `undefined` are falsey in Javascript, so you could just use
```
if (!snippet || snippet === this.state.prevSnippet) {
   return;
}
```